daras_ai_v2/language_model.py

```diff
@@ -1964,11 +1964,27 @@ def entry_to_prompt_str(entry: ConversationEntry) -> str:
 
 
 def format_chat_entry(
-    *, role: str, content: str, images: list[str] = None
+    *,
+    role: str,
+    content_text: str,
+    input_images: typing.Optional[list[str]] = None,
+    input_audio: typing.Optional[str] = None,
+    input_documents: typing.Optional[list[str]] = None,
+    render_input_urls: typing.Optional[bool] = False,
 ) -> ConversationEntry:
-    if images:
+
+    input_urls = []
+    if input_images and not render_input_urls:
+        input_urls.append(f"Image URLs: {', '.join(input_images)}")
+    if input_audio and not render_input_urls:
+        input_urls.append(f"Audio URL: {input_audio}")
+    if input_documents and not render_input_urls:
+        input_urls.append(f"Document URLs: {', '.join(input_documents)}")
+
+    content = content_text + ("\n" + "\n".join(input_urls) if input_urls else "")
```

Comment:
you don't need to do the multiple concatenation and then a join() + an inline if-condition. it could have all been folded into one join()